crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs

```diff
@@ -1997,15 +1997,24 @@ impl Analyzer<'_, '_> {
                 //
                 match *rhs.normalize() {
                     Type::Tuple(Tuple { elems: ref rhs_elems, .. }) => {
-                        // TODO: Handle Type::Rest
-
                         if elems.len() < rhs_elems.len() {
+                            if elems.len() > 0 {
```

Comment:
I think we should handle non-rest elements, too

Reply:
Is there a test case? Im not quite sure what other elements to test

I can't think of a case when the lengths won't match and the last element won't be a spread

Reply:
With current code we only check for last element.

```ts
declare let a: [string, number, ...boolean]
declare let b: [string, ...boolean]
declare let c: [string, number, boolean, boolean]

// Should fail
a = b
b = a

// Should success
a = c

// Should fail because of ["", 1]
c = a
```

Reply:
Awesome, got it :+1: